app/main/util/date_validator.py

```diff
@@ -17,33 +17,40 @@ def validate_dates(args, browse_consignment=False):
         errors["date_filter_field"] = (
             "Select either ‘Date of record’ or ‘Record opening date’"
         )
-        return None, None, None, None, None, None, errors
+        return None, None, None, None, None, None, errors, error_fields
 
     check_future_date = date_filter_field != "opening_date"
 
-    from_day, from_month, from_year, date_from_errs = validate_date(
-        args.get("date_from_day"),
-        args.get("date_from_month"),
-        args.get("date_from_year"),
-        "‘Date from’",
-        check_future_date,
+    from_day, from_month, from_year, date_from_errs, error_field = (
+        validate_date(
+            args.get("date_from_day"),
+            args.get("date_from_month"),
+            args.get("date_from_year"),
+            "‘Date from’",
+            check_future_date,
+        )
     )
-    if not date_from_errs and (from_day or from_month or from_year):
+    if not date_from_errs:
         from_day, from_month, from_year = complete_date(
             from_day, from_month, from_year, "from", check_future_date
         )
+    else:
+        error_fields.append(error_field)
 
-    to_day, to_month, to_year, date_to_errs = validate_date(
+    to_day, to_month, to_year, date_to_errs, error_field = validate_date(
         args.get("date_to_day"),
         args.get("date_to_month"),
         args.get("date_to_year"),
         "‘Date to’",
         check_future_date,
     )
-    if not date_to_errs and (to_day or to_month or to_year):
+
+    if not date_to_errs:
```

Comment:
Why did we remove this condition? We shouldn't complete the date if we dont have any of to_day, to_month or to_year

Reply:
"and (to_day or to_month or to_year)" was not executed as a conditional statement and returned True or False, if 'not date_to_errs' is suffice enough to check entered date values are correct and return the correct date